File: backend/src/main/java/com/urjc/backend/model/POD.java
package com.urjc.backend.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@Entity
public class POD {

    @EmbeddedId
    private PODKey id = new PODKey();

    @ManyToOne
    @MapsId("subjectId")
    @JoinColumn(name = "subject_id")
    private Subject subject;

    @ManyToOne
    @MapsId("courseId")
    @JoinColumn(name = "course_id")
    private Course course;

    @ManyToOne
    @MapsId("teacherId")
    @JoinColumn(name = "teacher_id")
    private Teacher teacher;

    @Column(nullable = false)
    private Integer chosenHours;
}

File: frontend/src/app/models/pod.ts
import { Course } from "./course";
import { SubjectModel } from "./subject";
import { Teacher } from "./teacher";

export class Pod {
    constructor(
        public id:number|any,
        public subject:SubjectModel|any,
        public course: Course|any,
        public teacher: Teacher|any,
        public chosenHours:number|any
    ) {

    }
}
